pkg/env/certificateRefresher.go

```diff
@@ -0,0 +1,114 @@
+package env
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"sync/atomic"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type CertGetter func(context.Context, string) (*rsa.PrivateKey, []*x509.Certificate, error)
```

Comment:
I have designed the Certificate refresher to not use the `keyvault.Manager` to limit the amount of dependencies and to simplify the testing.

With only function passed unit testing is dead simple,contrary to the whole mocked interface which would require quite a lot of unnecessary boilerplate code. 

I honestly believe it not necessary to pass the whole manager where only a certificate is of interest.

Reply:
I appriciate the intention to limit what we pass into the refresher, but I have few issues with `CertGetter`. 

1. It comes together with a custom test code which is hard to understand. The more custom code we have in our tests the harder it will be for us to maintain our tests. I agree that that GoMock will require some boilerplate, but it is likely that it will be less the amount of code `getterFactory` introduces. Also GoMock is a well known library which we use across our codebase: you learn it once and you can understand tests across this codebase.
2. Current custom test code doesn't test few important bits:
    * It doesn't test how many times we call `CertGetter` which is important here because we run it in a loop.
    * It doesn't test what arguments we pass into `CertGetter`

Reply:
+1 on using gomock and otherwise conforming to the code and naming patterns in the repo.